elm/Swole/Components/MovementInput.elm

```diff
@@ -36,36 +35,30 @@ update msg movements =
         NoOp ->
             (movements, Cmd.none)
 
-        MovementChanged (idx, m) ->
+        MovementChanged idx m ->
             let
-                (newIdx, newMovements) = splitMovements (idx, m)
+                (newIdx, newMovements) = splitMovements idx m
                 (before, after) = movementsAround idx movements
                 allMovements = before ++ newMovements ++ after
             in
-                (List.enumerated allMovements, updateFocus newIdx)
+                (allMovements, updateFocus newIdx)
 
         AddMovement ->
             let
-                rawMovements = List.map Tuple.second movements
-                idx = List.length rawMovements
+                idx = List.length movements
```

Comment:
Having not tried this locally, I can't guarantee this is the case, buuuuut... would this break? Specifically, if you're building indices off of the length of the list, deleting and then adding will cause indices to be captured in a duplicate fashion?

Reply:
Nope! This PR is specifically around removing that capture of the index. We're never capturing anything anymore. The index for modification/deletion is _always_ set during `view` and is never set anywhere else. So the end of the list will _always_ be the actual end of the list, and deleting an item at an index will _always_ correspond to the correct index.

That being said: I need to look into testing elm applications. This logic is tricky enough that I think I'd like to have some proper testing around it.